default_warehouse_from_sale_team/models/ir_rule.py

```diff
@@ -0,0 +1,25 @@
+from odoo import _, models
+from odoo.exceptions import AccessError
+
+
+class IrRule(models.Model):
+    _inherit = "ir.rule"
+
+    def _make_access_error(self, operation, records):
+        """Ensure that the selected journal is accessible for the user's team/branch.
+        If no team are defined for the journal, access is open to all users.
+        """
+        if records and records._name == "account.move":
+            journals = records.mapped("journal_id").sudo()
+            if journals:
+                journal_names = ", ".join(journals.mapped("display_name"))
+                records.invalidate_recordset()
```

Comment:
You should probably call the super so it takes care of logging the mssage and invalidating recordsets.